tests/validators/test_function.py

```diff
@@ -942,3 +942,32 @@ def f_w(v: Any, handler: core_schema.ValidatorFunctionWrapHandler, info: core_sc
     assert v.validate_json('1') == 1
     assert calls == ['json']
     calls.clear()
+
+
+def test_reprs() -> None:
+    reprs: List[str] = []
+
+    def sample_repr(v: Any, info: core_schema.ValidationInfo) -> Any:
+        reprs.append(repr(info))
+        return v
+
+    v = SchemaValidator(
+        core_schema.chain_schema(
+            [
+                core_schema.general_plain_validator_function(sample_repr),
+                core_schema.field_plain_validator_function(sample_repr, field_name='x'),
+            ]
+        )
+    )
+
+    class Foo:
+        def __repr__(self) -> str:
+            return 'This is Foo!'
+
+    v.validate_python(Foo())
+
+    # insert_assert(reprs)
+    assert reprs == [
+        'ValidationInfo(config=None, context=None)',
+        "FieldValidationInfo(config=None, context=None, field_name='x')",
+    ]
```

Comment:
It might be nice to also have the case with `data` present, though I'll forgive you for not having it 😂